mongodb-forks/drone-helm3
# Add new convert command based on helm 2to3 code

- Added a new convert command (mode) to use the feature independently
- Adjust env.Config to accomodate required settings
- Run convert before upgrade if migration flag is enabled
- Update go.mod and go.sum accordingly

I've followed the same development pattern as the original repo, this should be mergable upstream if they want to accept this feature, that's also the reason why I decided to keep the `github.com/pelotech/drone-helm3` signature in this PR. After comments I'll most likely rebase to have only one clean commit.

There's some basic unit testing for one section of the new convert code, speficially the `v3ReleaseFound` function. Adding testing for the Execute() method is not possible, or at least not as is written now, but that's because the 2to3 signature for the `cmd.Convert` command requires passing a kubeconfig, that's not easy to mock. I'd have been better if it receives a kubernetes clientset but it doesn't. In order to test it properly, we would need to rewrite it completely and then we would loose the advantage of using it in the first place.

General process to test would be:
- Install new release with `helm2`
- Run convert command. This could be with docker or drone exec if a pipeline exists
- Verify that the v2 Release is not present and V3 is.
